advanced_topics.qmd

```diff
@@ -266,15 +277,359 @@ pums_person |>
+  summarise(
+    work_hours_mean = weighted.mean(hours_worked, w = PWGTP, na.rm = TRUE),
+    work_hours_predicted = weighted.mean(predicted, w = PWGTP),
+  )
+
+plot <- ggplot(modelled_data) +
+  aes(
+    x = factor(
+      edu,
+      levels = c(
+        "Less than High school",
+        "High school",
+        "Some college",
+        "College or higher"
+      )
+    ),
+    y = hours_worked,
+    weight = PWGTP
+    ) +
+  geom_violinhalf(scale = "width") +
+  layer(
+    geom = "point",
+    stat = "identity",
+    position = position_nudge(x = -0.08),
+    data = modelled_data_summary,
+    mapping = aes(x = edu, y = work_hours_mean, weight = NULL),
+    params = list(shape = 108, size = 6)
+  ) +
+  layer(
+    geom = "point",
+    stat = "identity",
+    position = position_nudge(x = -0.08),
+    data = modelled_data_summary,
+    mapping = aes(x = edu, y = work_hours_predicted, weight = NULL),
+    params = list(shape = 18, size = 4)
+  ) +
+  facet_grid(
+    cols = vars(self_employed),
+    labeller = as_labeller(c("TRUE" = "self-employed", "FALSE" = "not self-employed"))
+  ) +
+  coord_flip() +
+  theme_minimal() +
+    labs(
+      title = "Hours worked per week",
+      subtitle = "by employment type in 2018, Alaska\nvertical line is weighted mean, diamond model prediciton",
+      x = "education",
+      y = "hours worked per week"
+    )
+```
+
+::: {.content-visible when-format="html"}
+```{r}
+#| label: plot_predictions_plot
+#| warning: false
+#| echo: false
+plot
+```
+:::
+
+::: {.content-visible when-format="pdf"}
+```{r}
+#| label: plot_predictions_plot-pdf
+#| warning: false
+#| echo: false
+#| fig-height: 9
+#| fig-cap: "Hours worked per week by employment type in 2021 for Alaska. The density curve is the raw data, the line at the bottom the weighted mean and the diamond at the bottom is the weighted mean of the model predictions."
+plot
+```
+:::
+
+We are slightly off in most categories, but especially far off for high school and some college educated workers where we predict much lower working hours.
+We can also re-use this UDF for any subset, including predicting on all of the US in 2021.
+
+```{r}
+#| label: predict-commute-time-all-us
+modelled_data <- open_dataset("data/person/") |>
+  filter(year == 2021) |>
+  filter(AGEP >= 16, AGEP < 65, !is.na(COW)) |>
+    transmute(
+      self_employed = str_detect(as.character(COW), "Self-employed"),
+      edu = case_when(
+        grepl("bachelor|associate|professional|doctorate|master", SCHL, TRUE) ~
+          "College or higher",
+        grepl("college", SCHL, TRUE) ~
+          "Some college",
+        grepl("High school graduate|high school diploma|GED", SCHL, TRUE) ~
+          "High school",
+        TRUE ~ "Less than High school"
+      ),
+      hours_worked = WKHP,
+      PWGTP = PWGTP,
+      predicted = predict_work_hours(edu, self_employed),
+    ) |>
+  collect()
+
+modelled_data |> head()
+```
+
+And how good were our predictions?
```

Comment:
After the plot, we should add a paragraph that answers this question.

Reply:
Done. This actually _was here_ it was just in the wrong place (up about Alaska) 🤦 

I've also added a bit more narrative to these chunks